tests/support/function-mock.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Nectary;
+
+$mock_dictionary = [];
+$mock_counts = [];
+
+/**
+ * Function mocker. Use to mock php functions
+ */
+function create_function_mock( $unit_reference, $name, $times = 1 ) {
+  global $mock_dictionary;
+  global $mock_counts;
+
+  $mock_counts[ $name ] = $times;
+
+  $mock = $unit_reference->getMockBuilder( 'Object' )
+  ->setMethods( [ $name ] )
+  ->getMock();
+
+  $method = $mock->expects( $unit_reference->exactly( $times ) )
+  ->method( $name );
+
+  $mock_dictionary[ $name ] = $mock;
+
+  return $method;
+}
+
+function __perform( $name, $callback ) {
+  global $mock_dictionary;
+  global $mock_counts;
+
+  if ( array_key_exists( $name, $mock_dictionary ) ) {
+    $mock_counts[ $name ]--;
+    $value = $callback( $mock_dictionary[ $name ] );
+
+    if ( $mock_counts[ $name ] < 1 ) {
+      unset( $mock_dictionary[ $name ] );
+      unset( $mock_counts[ $name ] );
+    }
+
+    return $value;
+  }
```

Comment:
Forgot to add a failback. If not one is mocking the function, the real function should be called.